tool/settings.re

```diff
@@ -95,6 +95,7 @@ struct Settings {
     minimap_enabled: bool,
     minimap_size: float,
     minimap_alpha: float,
+    minimap_position: string,
```

Comment:
This should be an enum instead of a string. When storing the settings, it should be converted to a string, and when loading the settings, it should be converted back to the enum (using a match on the string).

The advantage of using an enum here is:
* You have one point in the code where all possible different values are defined.
* When you add or remove one of the possible values, all matches on that enum will produce an error as not all cases are handled. With the current string approach, you need to manually make sure that you modified every single `match` where the enum was used, and if you forget one, you get a runtime error instead of a compilation error, with the latter being easier to find and fix.
* Matches become exhaustive and don't need a catch-all case, reducing boilerplate and semantically unreachable code.